### DISCUSSION TITLE
new windows builds

### DISCUSSION THREAD

**Discussion by maddes8cht** [ggml-org/llama.cpp#1313] [Q&A] [answered]
First of all thanks for the new windows builds.
Now as there are four new builds, is there some information which one to choose or what the different builds mean?
There are the cudart builds  and the others. Whats the difference?

Then each of them comes as a cu11.7.1 and a cu12.0.1 build.
Again, what's the difference, what does that mean?

Can someone give some clarification please?

**Comment by SlyEcho** [comment, marked as answer]
This is work-in-progress https://github.com/ggerganov/llama.cpp/pull/1271 not yet merged.

These are all CUDA builds, for Nvidia GPUs, different CUDA versions and also for people that don't have the runtime installed, big zip files that include the CUDA .dll files.

**Reply by Green-Sky** [reply]
@maddes8cht depending on which nvidia driver and/or cuda runtime you have installed, you can use the one that is best suited.
cu12.1.0 -> built against cuda 12.1.0 and requires at least driver version 531.14
cu11.7.1 -> built against cuaa 11.7.1 requires version >=516.94

the `cudart-llama-bin-win-cu12.1.0-x64.zip` files contain the cuda runtime .dlls that you need to run it. but only if you did not install the cuda runtime from nvidia anyway.
I might need to rename them to clarify this. any suggestions?



@SlyEcho you could do a test release for your pr too, so we have atleast some binaries out there. :)

**Reply by mikeggh** [reply]
it is a little confusing, maybe call the cudart (higher megabyte release) something like portable?.. people should be able to reference that easily..
